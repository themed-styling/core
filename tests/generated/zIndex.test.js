import 'jest-styled-components'

import { zIndex } from '../../src/index.js'

import { testAllOn } from '../utilities.js'

testAllOn(zIndex, 'zIndex')
