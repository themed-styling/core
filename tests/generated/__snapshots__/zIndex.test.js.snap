// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zIndex renders 0 1`] = `
Array [
  .c0 {
  z-index: 0;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: 0;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: 0;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: 0;
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders function 1`] = `
Array [
  .c0 {
  z-index: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: has;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: has;
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders invalid theme path 1`] = `
Array [
  .c0 {
  z-index: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: invalid.path.1;
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders negative number 1`] = `
Array [
  .c0 {
  z-index: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: -42365;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: -42365;
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders object [breakpoints] 1`] = `
Array [
  @media screen and (min-width:November) {
  .c0 {
    z-index: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    z-index: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {
  .c0 {
    z-index: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    z-index: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    z-index: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    z-index: November;
  }
}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    z-index: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    z-index: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    z-index: November;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders object [breakpoints] 2`] = `
Array [
  @media May {
  .c0 {
    z-index: July;
  }
}

@media April {
  .c0 {
    z-index: August;
  }
}

@media January {
  .c0 {
    z-index: September;
  }
}

@media April {
  .c0 {
    z-index: November;
  }
}

@media June {
  .c0 {
    z-index: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {
  .c0 {
    z-index: July;
  }
}

@media April {
  .c0 {
    z-index: August;
  }
}

@media January {
  .c0 {
    z-index: September;
  }
}

@media April {
  .c0 {
    z-index: November;
  }
}

@media June {
  .c0 {
    z-index: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    z-index: January;
  }
}

@media April {
  .c0 {
    z-index: December;
  }
}

@media January {
  .c0 {
    z-index: April;
  }
}

@media April {
  .c0 {
    z-index: May;
  }
}

@media June {
  .c0 {
    z-index: June;
  }
}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    z-index: January;
  }
}

@media April {
  .c0 {
    z-index: December;
  }
}

@media January {
  .c0 {
    z-index: April;
  }
}

@media April {
  .c0 {
    z-index: May;
  }
}

@media June {
  .c0 {
    z-index: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders postitive number 1`] = `
Array [
  .c0 {
  z-index: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: 42365;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: 42365;
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders string 1`] = `
Array [
  .c0 {
  z-index: July;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: July;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: June;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: June;
}

<div
    className="c0"
  />,
]
`;

exports[`zIndex renders valid theme path 1`] = `
Array [
  .c0 {
  z-index: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  z-index: DarkSlateBlue;
}

<div
    className="c0"
  />,
]
`;
